packages/utilities/src/utils/useWindow.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useState, useEffect } from 'react';
+
+export const useWindow = <T = Window>(windowObject?: T) => {
+  const [controlledWindow, setControlledWindow] = useState<T | Window>();
+
+  /**
+   * Only reference `window` after initial render to support SSR environments
+   */
+  useEffect(() => {
+    if (windowObject) {
+      setControlledWindow(windowObject);
+    } else {
+      setControlledWindow(window);
```

Comment:
I think this is an awesome hook to have for `container-utilities`, however it has the potential to cause some confusion when SSR is ran. Without `environment` being defined, which is passed into the `useWindow` hook as `windowObject` (which is optional), this would throw an exception. You can try this using:

```
/**
 * @jest-environment node
 */
```
at the top of the `useWindow.spec.tsx` test file. Could we go for a safer alternative with `else if (typeof window !== 'undefined') { setControlledWindow(window); } else { setControlledWindow({}); }`?

Reply:
IMO, we would safely check for the native `window` first:

```
if (typeof window !== 'undefined') {
  setControlledWindow(window);
} else {
  setControlledWindow(windowObject || {});
}
```

Reply:
Fair points. I wouldn't wish to pass an empty object however but rather an exception if `typeof window` is undefined and `windowObject` is undefined as that should not be allowed as hook containers depend on event listeners on a window or window-like object. Many of these objections should also be looked into for [`useDocument` in React Components](https://github.com/zendeskgarden/react-components/blob/main/packages/theming/src/utils/useDocument.ts) as that hook was the prior art for this one.

Reply:
Upon further reflection, I think `windowObject` should be required argument in the args of `useWindow`. In the case of no `window` available, we enforce our hooks to create a mock window like object for safety. This way we do not need to add an exception.

Reply:
To be fair, the public hook API for `useWindow` belongs in `@zendeskgarden/react-theming` – along with `useDocument` – where `window` is an object stored on the `theme`. This container should simply receive the `window` (`windowObj`, `environment`, whatever we decide to call it) as a prop and act on it as if the consumer provided a valid object. This is consistent with other `react-containers` APIs.

Reply:
Per offline convo: Removed `useWindow` in favor of a required `environment` variable in https://github.com/zendeskgarden/react-containers/pull/405/commits/4d40073147be6bc368adaa4cd9a13ff55f460f39